guides/keras_nlp/transformer_pretraining.py

```diff
@@ -0,0 +1,481 @@
+classification dataset and our "end goal". This dataset is often used to benchmark
+language models. 
+- [WikiText-103](https://paperswithcode.com/dataset/wikitext-103): A medium sized
+collection of featured articles from English wikipedia, which we will use for
+pre-training.
+
+Finally, we will download a WordPiece vocabulary, to do sub-word tokenization later on in
+this guide.
+"""
+
+"""shell
+!pip install -q keras-nlp
+
+# Download pretraining data.
+!curl -O https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-103-raw-v1.zip
+!unzip wikitext-103-raw-v1.zip
+
+# Download finetuning data.
+!curl -O https://dl.fbaipublicfiles.com/glue/data/SST-2.zip
+!unzip SST-2.zip
+
+# Download vocabulary data.
+!curl -O
+https://storage.googleapis.com/tensorflow/keras-nlp/examples/bert/bert_vocab_uncased.txt
+"""
+
+"""
+### Define hyperparameters
+
+Next up we can import `keras_nlp`, `tensorflow` and `keras`.
+
+A simple thing we can do here is to enable [mixed
+percision](https://keras.io/api/mixed_precision/), which will speed up training by
+running most of our computations with 16 bit (instead of 32 bit) floating point numbers.
+Training a transformer can take a while, so it is important to pull out all the stops for
+faster training!
+"""
+
+import keras_nlp
+import tensorflow as tf
+from tensorflow import keras
+
+policy = keras.mixed_precision.Policy("mixed_float16")
+keras.mixed_precision.set_global_policy(policy)
+
+"""
+Next we define some hyperparameters we will use during training.
+"""
+
+# Preprocessing params.
+PRETRAINING_BATCH_SIZE = 64
+FINETUNING_BATCH_SIZE = 32
+SEQ_LENGTH = 128
+MASK_RATE = 0.2
+PREDICTIONS_PER_SEQ = 25
+
+# Model params.
+NUM_LAYERS = 4
+MODEL_DIM = 256
+INTERMEDIATE_DIM = 1024
+NUM_HEADS = 4
+DROPOUT = 0.1
+NORM_EPSILON = 1e-5
+
+# Training params.
+PRETRAINING_LEARNING_RATE = 2e-4
+PRETRAINING_EPOCHS = 10
+FINETUNING_LEARNING_RATE = 2e-5
+FINETUNING_EPOCHS = 3
+
+"""
+### Load data
+
+We load our data with [tf.data](https://www.tensorflow.org/guide/data), which will allows
+us to define input pipelines for tokenizing and preprocessing text.
+"""
+
+# Load SST-2.
+sst_train_ds = tf.data.experimental.CsvDataset(
+    "SST-2/train.tsv", [tf.string, tf.int32], header=True, field_delim="\t"
+).batch(FINETUNING_BATCH_SIZE)
+sst_val_ds = tf.data.experimental.CsvDataset(
+    "SST-2/dev.tsv", [tf.string, tf.int32], header=True, field_delim="\t"
+).batch(FINETUNING_BATCH_SIZE)
+
+# Load wikitext-103 and filter out short lines.
+wiki_train_ds = (
+    tf.data.TextLineDataset("wikitext-103-raw/wiki.train.raw")
+    .filter(lambda x: tf.strings.length(x) > 100)
+    .batch(PRETRAINING_BATCH_SIZE)
+)
+wiki_val_ds = (
+    tf.data.TextLineDataset("wikitext-103-raw/wiki.valid.raw")
+    .filter(lambda x: tf.strings.length(x) > 100)
+    .batch(PRETRAINING_BATCH_SIZE)
+)
+
+# Take a peak at the sst-2 dataset.
+list(sst_train_ds.unbatch().take(4).as_numpy_iterator())
```

Comment:
Anything you want to print should be printed explicitly; this file should work as a Python script. Consider iterating over the dataset and calling `print()` on each sample